Library/Homebrew/cmd/bottle.rb

```diff
@@ -300,72 +301,86 @@ def bottle_formula(f)
     puts "./#{filename}"
     puts output
 
-    if ARGV.include? "--rb"
-      File.open("#{filename.prefix}.bottle.rb", "w") do |file|
-        file.write("\# #{f.full_name}\n")
-        file.write(output)
+    if ARGV.include? "--json"
+      json = {
+        f.full_name => {
+          "formula" => {
+            "pkg_version" => f.pkg_version.to_s,
+            "path" => f.path.to_s.strip_prefix("#{HOMEBREW_REPOSITORY}/"),
+          },
+          "bottle" => {
+            "disabled?" => f.bottle_disabled?,
+            "disable_reason" => f.bottle_disable_reason,
+            "root_url" => bottle.root_url,
+            "prefix" => bottle.prefix,
+            "cellar" => bottle.cellar.to_s,
+            "revision" => bottle.revision,
+            "tags" => {
+              Utils::Bottles.tag.to_s => {
+                "filename" => filename.to_s,
+                "sha256" => sha256,
+              },
+            }
+          },
+          "bintray" => {
+            "package" => Utils::Bottles::Bintray.package(f.full_name),
+            "repository" => Utils::Bottles::Bintray.repository(f.tap),
+          },
+        },
+      }
+      File.open("#{filename.prefix}.bottle.json", "w") do |file|
+        file.write Utils::JSON.dump json
       end
     end
   end
 
-  module BottleMerger
-    def bottle(&block)
-      instance_eval(&block)
-    end
-  end
-
   def merge
+    require "formulary"
+    Formulary.class_eval { define_method(:load_formula) { raise "Loaded formula!" } }
+
     write = ARGV.include? "--write"
-    keep_old = ARGV.include? "--keep-old"
-
-    merge_hash = {}
-    ARGV.named.each do |argument|
-      bottle_block = IO.read(argument)
-      formula_name = bottle_block.lines.first.sub(/^# /, "").chomp
-      merge_hash[formula_name] ||= []
-      merge_hash[formula_name] << bottle_block
+
+    bottles_hash = ARGV.named.reduce({}) do |hash, json_file|
+      hash.merge! Utils::JSON.load(IO.read(json_file))
     end
 
-    merge_hash.each do |formula_name, bottle_blocks|
+    bottles_hash.each do |formula_name, bottle_hash|
       ohai formula_name
-      f = Formulary.factory(formula_name)
 
-      if f.bottle_disabled?
-        ofail "Formula #{f.full_name} has disabled bottle"
-        puts f.bottle_disable_reason
+      if bottle_hash["bottle"]["disabled?"]
+        ofail "Formula #{formula_name} has disabled bottle"
+        puts bottle_hash[:bottle][:disable_reason]
         next
       end
```

Comment:
This is a dead code because a bottle disabled formula will never generate json file in the first place for merging